acapy_plugin_toolbox/basicmessage.py

```diff
@@ -45,37 +46,38 @@
 
 
 async def setup(
-        session: ProfileSession,
+        context: InjectionContext,
         protocol_registry: ProblemReport = None
 ):
     """Setup the basicmessage plugin."""
     if not protocol_registry:
-        protocol_registry = session.inject(ProtocolRegistry)
+        protocol_registry = context.inject(ProtocolRegistry)
     protocol_registry.register_message_types(
         MESSAGE_TYPES
     )
 
-    event_bus = session.inject(EventBus)
+    event_bus = context.inject(EventBus)
     event_bus.subscribe(BASIC_MESSAGE_EVENT_PATTERN, basic_message_event_handler)
 
 
-async def basic_message_event_handler(context: EventContext, event: Event):
+async def basic_message_event_handler(profile: Profile, event: Event):
     """
     Handle basic message events.
 
     Send a notification to admins when messages are received.
     """
 
     msg: BasicMessageRecord = BasicMessageRecord.deserialize(event.payload)
-    msg.sent_time = context.message.sent_time
+    msg.state = BasicMessageRecord.STATE_RECV
```

Comment:
This got me thinking about naming convention of events. This event is called `"basicmessages"` but is only sent when a basic message is received. It'd probably be a pain to change it now, but might be nice to start working toward a naming convention for aca-py event topics where they have 2 parts, the domain/namespace and the action/name of the event. So in this case, the event topic would be `"basicmessage_received"`

Reply:
Agreed, I had similar thoughts. The main obstacle to namespacing events at the moment is backwards compatibility on the webhooks. The topic is used in the path when posted to the webhook url. Right now, we emit an event when a webhook is sent. When we invert that and make the webhook sending just another event subscriber, that would be a good point to add namespacing to the event topics and then the webhook listener can map back to the original topic it sent out (`basicmessage_recieved` becomes `basicmessages`). This would also allow us to simplify the event handlers and explicitly listen for a message implying the state instead of listen for everything then check state of the payload.